src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -411,15 +411,22 @@ uct_ib_mlx5_post_send(uct_ib_mlx5_txwq_t *wq,
     /* We don't want the compiler to reorder instructions and hurt latency */
     ucs_compiler_fence();
 
-    /* Advance queue pointer */
+    /*
+     * Advance queue pointer.
+     * We return the number of BBs the *previous* WQE has consumed, since CQEs
+     * are reporting the index of the first BB rather than the last. We have
+     * reserved QP space for at least UCT_IB_MLX5_MAX_BB to accommodate.
+     * */
     ucs_assert(ctrl == wq->curr);
-    wq->curr       = src;
-    wq->prev_sw_pi = wq->sw_pi;
-    wq->sw_pi      = sw_pi;
+    res_count       = wq->sw_pi - wq->prev_sw_pi;
+    wq->curr        = src;
+    wq->prev_sw_pi += res_count;
+    ucs_assert(wq->prev_sw_pi == wq->sw_pi);
```

Comment:
minor: looks excessive, since it would never fail. 
`res_count` is a difference between `sw_pi` and `prev_sw_pi`. Thus `prev_sw_pi + res_count` will always be the same as `sw_pi`

Reply:
yeah, it should serve more as a documentation that we really set prev_sw_pi to sw_pi